File: src/leetcode/counting-words-with-a-given-prefix/counting-words-with-a-given-prefix.js
/**
 * Leetcode #2185
 * Difficulty: Easy
 * https://leetcode.com/problems/counting-words-with-a-given-prefix/
 */
/**
 * @param {string[]} words
 * @param {string} pref
 * @return {number}
 */
export const prefixCount = (words, pref) => {
  let count = 0;

  for (const word of words) {
    let isOk = true;

    for (let j = 0; j < pref.length; j += 1) {
      if (word[j] !== pref[j]) {
        isOk = false;
        break;
      }
    }

    if (isOk) {
      count += 1;
    }
  }

  return count;
};

// export const prefixCount = (words, pref) => words.filter(word => word.slice(0, pref.length) === pref).length;

File: src/leetcode/counting-words-with-a-given-prefix/counting-words-with-a-given-prefix.rs
impl Solution {
    pub fn prefix_count(words: Vec<String>, pref: String) -> i32 {
        words
            .into_iter()
            .filter(|word| word.len() >= pref.len() && word[..pref.len()] == pref)
            .collect::<Vec<String>>()
            .len() as i32
    }
}